src/generator/documentation.jl

```diff
@@ -14,15 +9,19 @@ function print_documentation(io::IO, cursor::CLCursor, indent, options, members:
     ids = get(options, "DAG_ids", Dict())
 
     if extract_c_comment_style == "doxygen"
-        doc = format_doxygen(cursor, options, members)
+        doc = format_doxygen(node.cursor, options, members)
     elseif extract_c_comment_style == "raw"
-        doc = format_raw(cursor, members)
+        doc = format_raw(node.cursor, members)
     else
-        return
+        if (callback_documentation = get(options, "callback_documentation", nothing)) ≢ nothing
+            doc = callback_documentation(node)
+        else
+            return
+        end
     end
     
-    if show_c_function_prototype && cursor isa Clang.CLFunctionDecl
-        prototype = ["### Prototype", "```c", get_prototype(cursor), "```"]
```

Comment:
was this functional before ? (i.e. `get_prototype` signature states `node`, not `cursor`)

Reply:
cc @melonedo

Reply:
Not sure whether it works now, but if it confuses you, `node.cursor.cursor` and `node.cursor` both work for some functions. This function used to accept ExprNode so it was possible I did not adjust it.

Reply:
You tell me, should I restore to the previous (being cautious) i.e. `get_prototype(node.cursor)` instead ?

Reply:
I think we can fix it in this way:

```
get_prototype(node) = get_prototype(node.cursor)

function get_prototype(cursor::CLCursor)
    code = Clang.getSourceCode(cursor)
    code = replace(code, r"/\*.*?\*/|//.*?$"m=>"")
    code = replace(code, r"\s+"m=>" ")
    "$code;"
end
```

Reply:
Thanks, pushed in 3d792337c49678506c1f9ea559c039a2b697ee81.

Reply:
get_prototype is only called here (actually, it was factored out to shorten the function body), so you are free to make any change as long as it works.